test/e2e/autoscaling/horizontal_pod_autoscaling.go

```diff
@@ -104,20 +108,69 @@ var _ = SIGDescribe("[Feature:HPA] Horizontal pod autoscaling (scale resource: C
 			doNotScaleOnBusySidecar("rs", e2eautoscaling.KindReplicaSet, true, f)
 		})
 	})
+
+	ginkgo.Describe("CustomResourceDefinition", func() {
+		ginkgo.It("Should scale with a CRD targetRef", func() {
+			scaleTest := &HPAScaleTest{
+				initPods:                    1,
+				totalInitialCPUUsage:        150,
+				perPodCPURequest:            200,
+				targetCPUUtilizationPercent: 50,
+				minPods:                     1,
+				maxPods:                     2,
+				firstScale:                  2,
+				targetRef:                   e2eautoscaling.CustomCRDTargetRef(),
+			}
+			scaleTest.run("rc-light", e2eautoscaling.KindRC, f)
+		})
+	})
+})
+
+var _ = SIGDescribe("[Feature:HPA] Horizontal pod autoscaling (scale resource: custom metrics)", func() {
```

Comment:
Question: perhaps this test would be easier to add (less changes to HPAScaleTest etc.) to the e2e tests file for custom metrics (in this directory)?

Reply:
Good point. I moved into into custom metrics one and it works fine with some minor changes.